test/test_nestedtensor.py

```diff
@@ -43,6 +62,17 @@ def noncontiguous_to_padded_tensor(input, shape=None):
     return result
 
 class TestNestedTensor(TestCase):
+    # Helper function to assert 2 nested tensors are equal
+    def nt_equal(self, nt1, nt2):
+        self.assertEqual(nt1.dtype, nt2.dtype)
+        self.assertEqual(nt1.device, nt2.device)
+        ub1 = nt1.unbind()
+        ub2 = nt2.unbind()
+        self.assertEqual(len(ub1), len(ub2))
+        n = len(ub1)
+        for i in range(n):
+            self.assertEqual(ub1[i], ub2[i])
```

Comment:
this is copy / pasted from `TestNestedTensorDeviceType` isn't it? Can we move this logic into `self.assertEqual()` instead?

Reply:
Yeah this is a great call out I will upstream these changes in the NT case to asserEqual